### PR TITLE
Added rule to detect docker client in container

### PR DESCRIPTION
**Pull request by daviddetorres**
**What type of PR is this?**

/kind feature
/kind rule-create

**Any specific area of the project related to this PR?**

/area rules

**What this PR does / why we need it**:

The motivation of this new rule is to detect suspicious activity like seen in the Graboid threat (more details here: https://unit42.paloaltonetworks.com/graboid-first-ever-cryptojacking-worm-found-in-images-on-docker-hub/). 

There are rules for detection of miners, but not the scenario where the container is used to attack other vulnerable hosts via a docker client. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
The priority is set to WARNING, but I have doubts if it should be other. I don't have the experience enough to decide this part of the rule.

**Does this PR introduce a user-facing change?**:
NONE

```release-note
A new rule is added to detect the execution of the docker client in a container and logs it with WARNING priority. 
```

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -2595,6 +2595,14 @@
   priority: CRITICAL
   tags: [process, mitre_execution]
 
+# This rule detects teh execution of the docker client inside a container
```
Comment (palmerabollo) [inline rules/falco_rules.yaml]:
```suggestion
# This rule detects the execution of the docker client inside a container
```

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -2595,6 +2595,14 @@
   priority: CRITICAL
   tags: [process, mitre_execution]
 
+# This rule detects the execution of the docker client inside a container
+- rule: The docker client is executed in a container
+  desc: An event will trigger every time you run a docker client in a container
+  condition: evt.type = execve and evt.dir=< and container.id != host and proc.name in (docker_binaries)
```
Comment (Kaizhe) [inline rules/falco_rules.yaml]: can you make the changes to 

```
- list: k8s_client_binaries
   items: [docker, kubectl, crictl, etcdctl, calicoctl]
```
`- rule: K8s client tool executed inside a container`
`condition: spawned_process and container and proc.name in (k8s_client_binaries)`

` tags: [container, mitre_execution]`

Comment (daviddetorres) [reply rules/falco_rules.yaml]: Thanks for the comments, Kaizhe. I didn't realize the macros for spawned process and container. 

I'll modify the pull request and upload the changes.

Comment (daviddetorres) [reply rules/falco_rules.yaml]: Also, I need some help for adding DCO signoff to a previous commit of this PR... I resolved a code review and forgot to add it. How could I add the DCO to an already existing commit?
(Sorry for the newbie mistake...).

Comment (Kaizhe) [reply rules/falco_rules.yaml]: No, not a problem, I had this issue before :) take a look this one: https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits

File: rules/falco_rules.yaml [hunk 3]
```diff
@@ -2595,6 +2595,17 @@
...
   tags: [process, mitre_execution]
 
+# This rule detects the execution of the docker client inside a container
+- list: k8s_client_binaries
+  items: [docker, kubectl, crictl]
+
+- rule: The docker client is executed in a container
+  desc: An event will trigger every time you run a docker client in a container
+  condition: spawned_process and container and proc.name in (k8s_client_binaries)
+  output: "Docker or kubernetes client in container (user=%user.name %container.info parent=%proc.pname cmdline=%proc.cmdline)"
```
Comment (Kaizhe) [inline rules/falco_rules.yaml]: Minor: 
`ouptut: Docker or kubernetes client executed in container`

Comment (daviddetorres) [reply rules/falco_rules.yaml]: Thanks Kaizhe for your support and help with my first PR in Falco... I'll commit this changes that you suggest, but before I'd like to fix the signature problem that it is giving in the this commit. 

I followed the link that you suggested and I added a gpg key (verified) and also a "Signed-off-by" in the commit. Even though, the problem persist... Any clue about how to fix it?

Comment (Kaizhe) [reply rules/falco_rules.yaml]: can you try `rebase HEAD~3` and squash the one didn't commit?

File: rules/falco_rules.yaml (hunk 2 above)
Comment (Kaizhe) [reply rules/falco_rules.yaml]: Please rebase from `dev` too

File: rules/falco_rules.yaml (hunk 3 above)
Comment (daviddetorres) [reply rules/falco_rules.yaml]: It seems it worked at last!  Thanks again for the support :)

Commit pushed with your suggestions.

File: rules/falco_rules.yaml [hunk 4]
```diff
@@ -2599,3 +2599,9 @@
 # there if you want to enable them by adding to
 # falco_rules.local.yaml.
 
+- rule: The docker client is executed in a container
+  desc: Detect a k8s client tool executed inside a container
+  condition: spawned_process and container and proc.name in (k8s_client_binaries)
```
Comment (Kaizhe) [inline rules/falco_rules.yaml]: Did you accidentally delete the `k8s_client_binaries` list? And please move the rule above 
```
# Application rules have moved to application_rules.yaml. Please look
# there if you want to enable them by adding to
# falco_rules.local.yaml.
```

Comment (daviddetorres) [reply rules/falco_rules.yaml]: Thanks again Kaizhe. You are really being patience with me in this PR...

Comment (Kaizhe) [reply rules/falco_rules.yaml]: Thank you for your contribution! It is really appreciated!

**Comment by fntlnz** [comment]
Thanks for improving our ruleset @daviddetorres !!!

**Pull request merged**